packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,210 @@
+import { Component, Event, EventEmitter, Method, Prop, State } from '@stencil/core';
+import readableNumber from '../../utils/readable-number';
+import { shadeOrBlend } from '../../utils/styles';
+
+const OTHER_CATEGORY_COLOR = '#747474';
+const OTHER_CATEGORY_NAME = 'Other';
+const DEFAULT_BAR_COLOR = '#47DB99';
+
+@Component({
+  shadow: false,
+  styleUrl: './as-category-widget.scss',
+  tag: 'as-category-widget',
+})
+export class CategoryWidget {
+  @Prop() public categories: object[] = [];
+  @Prop() public defaultBarColor: string = DEFAULT_BAR_COLOR;
+  @Prop() public description: string;
+  @Prop() public heading: string;
+  @Prop() public showClearButton: boolean = false;
+  @Prop() public showHeader: boolean = true;
+  @Prop() public useTotalPercentage: boolean = false;
+  @Prop() public visibleCategories: number = Infinity;
+
+  @Event() public categoriesSelected: EventEmitter<string[]>;
+
+  @State() private selectedCategories: string[] = [];
+
+  @Method()
+  public getSelectedCategories() {
+    return this.selectedCategories;
+  }
+
+  @Method()
+  public clearSelection() {
+    if (!this.selectedCategories.length) {
+      return;
+    }
+
+    this.selectedCategories = [];
+    this._onCategoriesChanged();
+  }
+
+  public render() {
+    return [
+      this._renderHeader(),
+      this._renderCategoryList(),
+      this._renderFooter()
+    ];
+  }
+
+  private _renderHeader() {
+    if (!this.showHeader) {
+      return;
+    }
+
+    return [
+      <h2 class='as-subheader'>{this.heading}</h2>,
+      <p class='as-category-widget__description as-body'>{this.description}</p>,
+    ];
+  }
+
+  private _renderCategoryList() {
+    return <ul class='as-category-widget__list'>{this._renderCategories()}</ul>;
+  }
+
+  private _renderCategories() {
+    const moreCategoriesThanVisible = this.categories.length > this.visibleCategories;
+    const { categories, otherCategory } = this._parseCategories();
+    let otherCategoryTemplate;
+
+    const categoriesToRender =  moreCategoriesThanVisible
```

Comment:
I think we can avoid this ternary you can always do `categories.slice(0, this.visibleCategories)`, if the array is smaller it will return it with the same size, in fact I think we can just use `this._getVisibleCategories`